README.md

```diff
@@ -1,2 +1,213 @@
-# web-devops-roadmap
-DevOps Roadmap for web developers
+# Web DevOps Roadmap
+
+This repository contains initial DevOps roadmap for all the web developers. Anyone can start this training with their existing learning projects.
+
+
+# Guidelines
+- Before starting any practice, it's essential to conduct research and learn the necessary concepts.
+
+- Create a GitHub/GitLab repository for each Practical if one doesn't already exist.
+
+- Learn Basic DevOps Principles:
+  - Learn how to use Version Control effectively for your codebase as Git is Fundamental.
+  - Know how to write unit tests.
+  - Start by understanding the core DevOps concepts like continuous integration, continuous deployment, and automation.
+  - Learn some basic scripts to get a basic understanding of scripting language.
+  - Be aware of basic security practices like exposing env vars, Inbound/Outbound security rules, etc…
+
+- Some Basic Guidelines for CI/CD:
+  - Understand the CI/CD concepts and their significance with pipeline usage.
+  - Learn how to set up the basic CI/CD pipeline with the GitLab/GitHub-CI tool.
+  - All lint rules should be passed from the pipeline.
+  - Always use the latest versions of languages and project dependencies
+  - Understand how to manage project dependencies using Node modules, Composer, or Vendors. And integrate them into the pipeline.
```

Comment:
`Understand how to manage project dependencies using npm(node package manager), Yarn, Composer, or Vendors and integrate them into the pipeline.`